ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot

```diff
@@ -0,0 +1,60 @@
+*** Settings ***
+Documentation       Test suite for OpenShift Pipeline
+Resource            ../../../Resources/RHOSi.resource
+Resource            ../../../Resources/ODS.robot
+Resource            ../../../Resources/Common.robot
+Resource            ../../../Resources/Page/ODH/ODHDashboard/ODHDashboard.robot
+Library             DateTime
+Library             ../../../../libs/DataSciencePipelinesAPI.py
+Suite Setup         Data Science Pipelines Suite Setup
+Suite Teardown      RHOSi Teardown
+
+
+*** Variables ***
+${REDHAT_OPENSHIFT_PIPELINES_YAML}=            tests/Resources/Files/redhat-openshift-pipelines.yaml
+${DATA_SCIENCE_PIPELINES_APPLICATION_YAML}=    tests/Resources/Files/data-science-pipelines-sample.yaml
+${URL_TEST_PIPELINE_RUN_YAML}=                 https://raw.githubusercontent.com/opendatahub-io/data-science-pipelines-operator/main/tests/resources/dsp-operator/test-pipeline-run.yaml
+
+*** Test Cases ***
+Test and end-to-end pipeline workflow via API
+    [Documentation]    Create, run and double check the pipeline result. In the end, clean the pipeline resources.
+    [Tags]    Smoke
+    End to end pipeline workflow via API    ${OCP_ADMIN_USER.USERNAME}    ${OCP_ADMIN_USER.PASSWORD}    myproject1
+    End to end pipeline workflow via API    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    myproject2
+    # This will fail because there is still a JIRA - RHODS-7615
+    # End to end pipeline workflow via API    ${TEST_USER_3.USERNAME}    ${TEST_USER_3.PASSWORD}    myproject3
+
+*** Keywords ***
+End to end pipeline workflow via API
+    [Documentation]    Create, run and double check the pipeline result. In the end, clean the pipeline resources.
+    [Arguments]     ${username}    ${password}    ${project}
+    Create Pipeline Project    ${project}
+    Install DataSciencePipelinesApplication CR    ${project}
+    ${status}    Login Using User And Password    ${username}    ${password}    ${project}
+    Should Be True    ${status} == 200    DSP routing is working
+    ${pipeline_id}    Create Pipeline    ${URL_TEST_PIPELINE_RUN_YAML}
+    ${run_id}    Create Run    ${pipeline_id}
+    ${run_status}    Check Run Status    ${run_id}
+    Should Be True    '${run_status}' == 'Completed'    Run ends
```

Comment:
would it make sense to check the pipeline output, if there's any output? These lines are checking the status only, which is ok, but if there is an expected result of the pipeline (e.g., a number, a text, whatever) we may want to check it too.

Reply:
It will keep running the `Check Run Status`

```pyton
        while run_status != 'Completed':
            print(f"Checking run status: {run_status}")

```

Reply:
okay, but what about the produced results of the pipeline? The pipelines runs some code doing something, does that have an output to the user, rather than the pipeline status.